app_feup/lib/model/entities/exam.dart

```diff
@@ -93,4 +102,14 @@ class Exam {
   String toString() {
     return '''$subject - $year - $month - $day -  $begin-$end - $examType - $rooms - $weekDay''';
   }
+
+  static Map<String, String> getExamTypes() {
+    return _types;
+  }
+
+  static getExamTypeLong(String abr) {
+    if (abr == null) return '';
+    final Map<String, String> reversed = _types.map((k, v) => MapEntry(v, k));
+    return reversed[abr];
```

Comment:
This method can return `null`. Since line 111 avoids returning a null value, I think this is unwanted behaviour. A good fix would be replacing lines 112 and 113 with `return _types.keys.firstWhere((type) => _types[type] == abr, orElse: () => '');`.
Another good fix would be replacing line 113 with `return reversed[abr] ?? '';`.



Personally, I think it would be best to simply return `null` in both situations: when `abr` is `null`  and when `abr` is not a key of `_types`. This would allow the function to be used in other places more easily.

Therefore, the code would look something along the lines of
```suggestion
try {
  return _types.keys.firstWhere((type) => _types[type] == abr));
} catch (e) {
  return null;
}
```

Reply:
Done! Thanks!